saaj-ri/src/main/java/com/sun/xml/messaging/saaj/util/transform/EfficientStreamingTransformer.java

```diff
@@ -73,16 +74,33 @@
      * Underlying FI DOM serializer.
      */
     private Object m_fiDOMDocumentSerializer = null;
+
+    private static final AtomicBoolean TO_BE_LOGGED = new AtomicBoolean(true);
     
     private EfficientStreamingTransformer() {
         TransformerFactory tf = TransformerFactory.newInstance();
+        final boolean toBeLogged = TO_BE_LOGGED.compareAndSet(true, false);
         try {
             tf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
         } catch (TransformerConfigurationException e) {
-            LOGGER.log(Level.WARNING, "Factory [{0}] doesn't support secure xml processing!", new Object[] { tf.getClass().getName() } );
+            if (toBeLogged) {
```

Comment:
Why this condition is here? One can control logging via logging.properties file which can be tweaked as needed. The only condition I could accept on this place would be something like `LOGGER.isLoggable(Level.WARNING)`

Reply:
In wildfly-common this 'toBeLogged' notion was added to a utility class as a 'log it once, not every time newTransformer() is called' thing.

Reply:
But this is really a warning level message and user needs to know if this feature setting like this one works or not:
`
 tf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
`